Flaky `rpctest` test

**Summary:** 
<!-- Summarize the problem in a few sentences: -->

Level of flakiness of `go.thethings.network/lorawan-stack/pkg/util/rpctest` is higher than expected on https://github.com/TheThingsNetwork/lorawan-stack/commit/09111e2b4712c969bc3373bf7aef2f0f52a68887

**Steps to Reproduce:** 
<!-- How can we reproduce the problem? -->

1. Run CI a couple of times

<!-- Please attach (or link to) configuration files if applicable -->

**What do you see now?**
<!-- Please attach (or link to) screenshots and logs if applicable -->

```
--- FAIL: TestFooBarExampleServer (0.45s)
    --- FAIL: TestFooBarExampleServer/ClientStream (0.15s)
        doc.go:66: 
            /home/travis/gopath/src/go.thethings.network/lorawan-stack/pkg/util/rpctest/server_test.go:84
            Expected: 'Canceled'
            Actual:   'OK'
            (Should be equal)
FAIL
coverage: 45.1% of statements
FAIL	go.thethings.network/lorawan-stack/pkg/util/rpctest	0.492s
```

https://travis-ci.com/TheThingsNetwork/lorawan-stack/jobs/186935339

**What do you want to see instead?**
<!-- Please add some examples or mock-ups if applicable -->

Successful test completion

**How do you propose to implement this?**
<!--
If unsure, add the discussion label and (temporarily) assign the expert
If you cannot assign people, please @mention the experts
-->

Deflake the test

**Environment:**
<!-- Your environment: OS/Browser/Gateway/Device/...? Versions? IDs/EUIs? -->

Any

--- a/pkg/util/rpctest/server_test.go
+++ b/pkg/util/rpctest/server_test.go
@@ -80,7 +80,7 @@ func TestFooBarExampleServer(t *testing.T) {
 			err = stream.Send(&rpctest.Foo{Message: "foo"})
 			a.So(err, should.BeNil)
 			cancel()
-			_, err = stream.CloseAndRecv()
+			err = stream.RecvMsg(&rpctest.Bar{})
 			a.So(grpc.Code(err), should.Equal, codes.Canceled)
 		}
 

